client/src/core/client/stream/tabs/Comments/IllegalContentReportView/IllegalContentReportViewContainer.tsx

```diff
@@ -0,0 +1,458 @@
+  const setCommentID = useMutation(SetCommentIDMutation);
+  const createDSAReport = useMutation(CreateDSAReportMutation);
+  const { eventEmitter, window } = useCoralContext();
+  const [additionalComments, setAdditionalComments] = useState<
+    null | { id: string; url: string }[]
+  >(null);
+  const [submissionStatus, setSubmissionStatus] = useState<
+    { id: string; status: string; error?: any; url: string }[]
+  >([]);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const submissionID = useUUID();
+
+  const onShowAllComments = useCallback(
+    (e: MouseEvent<any>) => {
+      ViewFullDiscussionEvent.emit(eventEmitter, {
+        commentID: comment && comment.id,
+      });
+      void setCommentID({ id: null });
+      // TODO: remove view param too
+      e.preventDefault();
+    },
+    [comment, eventEmitter, setCommentID]
+  );
+
+  const showAllCommentsHref = useMemo(() => {
+    const url = window.location.href;
+    return getURLWithCommentID(url, undefined);
+  }, [window.location.href]);
+
+  const commentVisible = comment && isPublished(comment.status);
+
+  const onSubmit = useCallback(
+    async (input: FormProps, form: FormApi) => {
+      const statuses = [];
+      if (viewer && comment) {
+        if (additionalComments) {
+          for (const c of additionalComments) {
+            try {
+              await createDSAReport({
+                userID: viewer.id,
+                commentID: c.id,
+                lawBrokenDescription: input.lawBrokenDescription,
+                additionalInformation: input.additionalInformation,
+                submissionID,
+              });
+              statuses.push({ id: c.id, status: "success", url: c.url });
+            } catch (e) {
+              statuses.push({
+                id: c.id,
+                status: "error",
+                error: e,
+                url: c.url,
+              });
+            }
+          }
+        }
+        const url = getURLWithCommentID(story.url, comment.id);
+        try {
+          await createDSAReport({
+            userID: viewer.id,
+            commentID: comment.id,
+            lawBrokenDescription: input.lawBrokenDescription,
+            additionalInformation: input.additionalInformation,
+            submissionID,
+            commentRevisionID: comment.revision?.id,
+          });
+          statuses.push({ id: comment.id, status: "success", url });
+        } catch (e) {
+          statuses.push({ id: comment.id, status: "error", error: e, url });
+        }
+        setSubmissionStatus(statuses);
+        setIsSubmitted(true);
+      }
+    },
+    [
+      additionalComments,
+      viewer,
+      comment,
+      createDSAReport,
+      setSubmissionStatus,
+      setIsSubmitted,
+      submissionID,
+      story.url,
+    ]
+  );
+
+  if (isSubmitted) {
+    return (
+      <>
+        {/* TODO: Localize and update to add in any errors and style based on design */}
```

Comment:
the confirmation screen after the illegal content report is submitted is still all copy/design TBD, but this gives us the data we will need for it about which reports were successfully submitted or had errors